### PR TITLE
Small docsite fixes and upgrade Sphinx

### PR DESCRIPTION
**Pull request by mikehearn**
After this PR the only warnings are pygments not understanding all of kotlin's syntax.

### CODE REVIEW

File: docs/source/getting-set-up.rst [hunk 1]
```diff
@@ -171,8 +171,8 @@ The best way to check that everything is working fine is by taking a deeper look
 Next, you should read through :doc:`Corda Key Concepts <key-concepts>` to understand how Corda works.
 
 By then, you'll be ready to start writing your own CorDapps. Learn how to do this in the
-:doc:`Hello, World tutorial <hello-world-introduction>`. You may want to refer to the :doc:`API docs <api-index>`, the
+:doc:`Hello, World tutorial <hello-world-introduction>`. You may want to refer to the API docs, the
```
Comment (tlil) [inline docs/source/getting-set-up.rst]: `API documentation`

**Comment by shamsasari** [comment]
Can you rebase, there's more changes to the changelog.

File: docs/source/cipher-suites.rst [hunk 2]
```diff
@@ -52,51 +53,51 @@ underlying HSM device, but the default for dev-mode is Pure EdDSA (ed25519).
...
-|                         | studied and well understood area of cryptography. Thus, it  |     |                       |
-|                         | is assumed that there is a much lower chance of             |     |                       |
-|                         | breakthrough attacks on the underlying mathematical         |     |                       |
-|                         | problems. However, SPHINCS uses relatively big public keys, |     |                       |
-|                         | it is slower and outputs bigger signatures than EdDSA,      |     |                       |
-|                         | ECDSA and RSA algorithms.                                   |     |                       |
-+-------------------------+-------------------------------------------------------------+-----+-----------------------+
++-------------------------+---------------------------------------------------------------+-----+-------------------------+
+| Cipher suite            | Description                                                   | TLS | Default for             |
```
Comment (shamsasari) [inline docs/source/cipher-suites.rst]: Doesn't the header need these double-width column margins as well?

File: docs/source/changelog.rst [hunk 3]
```diff
@@ -436,7 +396,7 @@ Release 1.0
 
 * Vault query soft locking enhancements and deprecations
 
-  * removed original ``VaultService`` ``softLockedStates` query mechanism.
+  * removed original ``VaultService`` ``softLockedStates`` query mechanism.
   * introduced improved ``SoftLockingCondition`` filterable attribute in ``VaultQueryCriteria`` to enable specification of different soft locking retrieval behaviours (exclusive of soft locked states, soft locked states only, specified by set of lock ids)
 
 * Trader demo now issues cash and commercial paper directly from the bank node, rather than the seller node self-issuing
```
Comment (shamsasari) [inline docs/source/changelog.rst]: Line 549 below: ` ``legalIdentitiesAndCert``s `

File: docs/source/cipher-suites.rst (hunk 2 above)
Comment (mikehearn) [reply docs/source/cipher-suites.rst]: you'd think so, but I fiddled about with this for quite a while until the table rendered well. I'd rather not touch it now. It's *remarkably* sensitive to pointless details.

**Comment by mikehearn** [comment]
Thanks for the review!

**Pull request merged**